core/src/main/java/io/kestra/core/schedulers/AbstractScheduler.java

```diff
@@ -736,54 +745,54 @@ private void emitExecution(Execution execution, TriggerContext trigger) {
         }
     }
 
-    private boolean isExecutionNotRunning(FlowWithWorkerTrigger f) {
-        Trigger lastTrigger = f.getTriggerContext();
-
-        if (lastTrigger.getExecutionId() == null) {
-            return true;
-        }
-
-        Optional<Execution> execution = executionState.findById(lastTrigger.getTenantId(), lastTrigger.getExecutionId());
-
-        // executionState hasn't received the execution, we skip
-        if (execution.isEmpty()) {
-            if (lastTrigger.getUpdatedDate() != null) {
-                metricRegistry
-                    .timer(MetricRegistry.METRIC_SCHEDULER_EXECUTION_MISSING_DURATION, MetricRegistry.METRIC_SCHEDULER_EXECUTION_MISSING_DURATION_DESCRIPTION, metricRegistry.tags(lastTrigger))
-                    .record(Duration.between(lastTrigger.getUpdatedDate(), Instant.now()));
-            }
-
-            if (lastTrigger.getUpdatedDate() == null || lastTrigger.getUpdatedDate().plusSeconds(60).isBefore(Instant.now())) {
-                logService.logTrigger(
-                    f.getTriggerContext(),
-                    Level.WARN,
-                    "Execution '{}' is not found, schedule is blocked since '{}'",
-                    lastTrigger.getExecutionId(),
-                    lastTrigger.getUpdatedDate()
-                );
+    private void executionMonitor() {
+        try {
+            // Retrieve triggers with non-null execution_id from all corresponding virtual nodes
+            ZonedDateTime now = ZonedDateTime.now();
+            List<Trigger> triggers = this.triggerState.findByNextExecutionDateReadyButLockedTriggers(now);
+            if (CollectionUtils.isEmpty(triggers)) {
+                log.warn("executionMonitor triggers is empty, skip");
```

Comment:
```suggestion
                log.debug("executionMonitor triggers is empty, skip");
```

If there is nothing to monitor, it's not an issue right? So better to write a debug log ;)